pkg/kubelet/kubelet.go

```diff
@@ -144,6 +144,9 @@ const (
 	// Max amount of time to wait for the container runtime to come up.
 	maxWaitForContainerRuntime = 30 * time.Second
 
+	// Max amount of time to admit a single Pod, mostly accounting for device plugin allocation.
+	maxSinglePodAdmissionTimeout = 30 * time.Second
```

Comment:
```suggestion
	podAdmissionTimeout = 30 * time.Second
```

Reply:
> It would be best if maxSinglePodAdmissionTimeout were a parameter of kubelet.

Yes, it can be a configuration parameter as well, but it should be covered by the `PodAdmissionWithTimeout ` feature gate.